### PR TITLE
ConfigPackageUtility::ValidatePackageName(): always tolerate already existing packages

### PR DESCRIPTION
**Pull request by Al2Klimov**
... not to require migrating invalid ones.

### CODE REVIEW

**Comment by Al2Klimov** [comment]
## 9169c80 w/o c40b18e

```
➜  icinga2 git:(501b3e043) curl -ksSu root:123456 -H 'Accept: application/json' 'https://localhost:5665/v1/config/packages/foo:bar?pretty=1' -X POST
{
    "results": [
        {
            "code": 200,
            "package": "foo:bar",
            "status": "Created package."
        }
    ]
}
➜  icinga2 git:(501b3e043) curl -ksSu root:123456 -H 'Accept: application/json' 'https://localhost:5665/v1/config/stages/foo:bar?pretty=1' -X POST -d '{"files":{}}'
{
    "results": [
        {
            "code": 200,
            "package": "foo:bar",
            "stage": "1a5e7cb2-e933-4b33-ba2d-cab3dc4ba213",
            "status": "Created stage. Reload triggered."
        }
    ]
}
➜  icinga2 git:(501b3e043) curl -ksSu root:123456 -H 'Accept: application/json' 'https://localhost:5665/v1/config/stages/foo:bar/1a5e7cb2-e933-4b33-ba2d-cab3dc4ba213?pretty=1'
{
    "results": [
        {
            "name": "include.conf",
            "type": "file"
        },
        {
            "name": "startup.log",
            "type": "file"
        },
        {
            "name": "status",
            "type": "file"
        },
        {
            "name": "conf.d",
            "type": "directory"
        },
        {
            "name": "zones.d",
            "type": "directory"
        }
    ]
}
➜  icinga2 git:(501b3e043)
```

## 9169c80

```
➜  icinga2 git:(9169c805a) curl -ksSu root:123456 -H 'Accept: application/json' 'https://localhost:5665/v1/config/stages/foo:bar/1a5e7cb2-e933-4b33-ba2d-cab3dc4ba213?pretty=1'
{
    "error": 400,
    "status": "Invalid package name 'foo:bar'."
}
➜  icinga2 git:(9169c805a)
```

## 9169c80 + this PR

```
➜  icinga2 git:(f836422ab) curl -ksSu root:123456 -H 'Accept: application/json' 'https://localhost:5665/v1/config/stages/foo:bar/1a5e7cb2-e933-4b33-ba2d-cab3dc4ba213?pretty=1'
{
    "results": [
        {
            "name": "include.conf",
            "type": "file"
        },
        {
            "name": "startup.log",
            "type": "file"
        },
        {
            "name": "status",
            "type": "file"
        },
        {
            "name": "conf.d",
            "type": "directory"
        },
        {
            "name": "zones.d",
            "type": "directory"
        }
    ]
}
➜  icinga2 git:(f836422ab)
```

However, c40b18e stays intact:

```
➜  icinga2 git:(f836422ab) curl -ksSu root:123456 -H 'Accept: application/json' 'https://localhost:5665/v1/config/stages/fox:bar/1a5e7cb2-e933-4b33-ba2d-cab3dc4ba213?pretty=1'
{
    "error": 400,
    "status": "Invalid package name 'fox:bar'."
}
➜  icinga2 git:(f836422ab)
```

File: lib/remote/configpackageutility.cpp [hunk 1]
```diff
@@ -367,7 +378,17 @@ bool ConfigPackageUtility::ContainsDotDot(const String& path)
...
 
-bool ConfigPackageUtility::ValidateName(const String& name)
+bool ConfigPackageUtility::ValidatePackageName(const String& packageName)
+{
+	return ValidateFreshName(packageName) || PackageExists(packageName);
+}
+
+bool ConfigPackageUtility::ValidateStageName(const String& packageName, const String& stageName)
+{
+	return ValidateFreshName(stageName) || StageExists(packageName, stageName);
```
Comment (julianbrost) [inline lib/remote/configpackageutility.cpp:388]: Is this required for stages? Shouldn't a stage name always be a UUID chosen by Icinga and thus also be valid using the new validation function? According to https://icinga.com/docs/icinga-2/latest/doc/12-icinga2-api/#create-a-stage-upload-configuration, there's no option to influence the stage name.

**Review by julianbrost — COMMENTED** [review]
Using 2.12.5 to create a package with a name that should have been invalid:
```console
$ curl -isSku root:icinga -H 'Accept: application/json' -X POST 'https://localhost:5665/v1/config/packages/foo.bar?pretty=1'
HTTP/1.1 200 OK
Server: Icinga/v2.12.5
Content-Type: application/json
Content-Length: 148

{
    "results": [
        {
            "code": 200.0,
            "package": "foo.bar",
            "status": "Created package."
        }
    ]
}
```

Trying to create a similarly named package using the new version (bbc39a055 = 56f0687a40922ff2400c6513ec61c39c9b8b6bb3 + `git merge master`):
```console
curl -isSku root:icinga -H 'Accept: application/json' -X POST 'https://localhost:5665/v1/config/packages/foo.baz?pretty=1'
HTTP/1.1 400 Bad Request
Server: Icinga/v2.12.0-824-gbbc39a055
Content-Type: application/json
Content-Length: 70

{
    "error": 400,
    "status": "Invalid package name 'foo.baz'."
}
```

However, it remains possible to use the already existing package:
```console
curl -isSku root:icinga -H 'Accept: application/json' -X POST -d '{ "files": { "zones.d/global-templates/foo.bar.test.conf": "# empty"}, "pretty": true }' 'https://localhost:5665/v1/config/stages/foo.bar'
HTTP/1.1 200 OK
Server: Icinga/v2.12.0-824-gbbc39a055
Content-Type: application/json
Content-Length: 224

{
    "results": [
        {
            "code": 200,
            "package": "foo.bar",
            "stage": "3b388863-fda5-4e30-bd77-d2c7f976d2ad",
            "status": "Created stage. Reload triggered."
        }
    ]
}
```

**Review by julianbrost — APPROVED** [review]
Retesting with 2eebd6f = 57df803e35b6a02b25a25195452f16f78f1c23e6 + `git merge master`.

```console
$ curl -isSku root:icinga -H 'Accept: application/json' -X POST 'https://localhost:5665/v1/config/packages/foo.baz?pretty=1'
HTTP/1.1 400 Bad Request
Server: Icinga/v2.12.0-823-g2eebd6f4e
Content-Type: application/json
Content-Length: 70

{
    "error": 400,
    "status": "Invalid package name 'foo.baz'."
}
```

```console
$ curl -isSku root:icinga -H 'Accept: application/json' -X POST -d '{ "files": { "zones.d/global-templates/foo.bar.test.conf": "# empty"}, "pretty": true }' 'https://localhost:5665/v1/config/stages/foo.bar'
HTTP/1.1 200 OK
Server: Icinga/v2.12.0-823-g2eebd6f4e
Content-Type: application/json
Content-Length: 224

{
    "results": [
        {
            "code": 200,
            "package": "foo.bar",
            "stage": "2e201e68-23bc-4bf3-abcc-a3fd350bb1f3",
            "status": "Created stage. Reload triggered."
        }
    ]
}
```

Also, interacting with methods that validate stage names is still possible (i.e. stage names are still considered valid):
```console
$ curl -isSku root:icinga -H 'Accept: application/json' 'https://localhost:5665/v1/config/stages/foo.bar/3b388863-fda5-4e30-bd77-d2c7f976d2ad?pretty=1'
HTTP/1.1 200 OK
Server: Icinga/v2.12.0-823-g2eebd6f4e
Content-Type: application/json
Content-Length: 653

{
    "results": [
        {
            "name": "include.conf",
            "type": "file"
        },
        {
            "name": "startup.log",
            "type": "file"
        },
        {
            "name": "status",
            "type": "file"
        },
        {
            "name": "conf.d",
            "type": "directory"
        },
        {
            "name": "zones.d",
            "type": "directory"
        },
        {
            "name": "zones.d/global-templates",
            "type": "directory"
        },
        {
            "name": "zones.d/global-templates/foo.bar.test.conf",
            "type": "file"
        }
    ]
}
$ curl -isSku root:icinga -H 'Accept: application/json' -X DELETE 'https://localhost:5665/v1/config/stages/foo.bar/3b388863-fda5-4e30-bd77-d2c7f976d2ad?pretty=1'
HTTP/1.1 200 OK
Server: Icinga/v2.12.0-823-g2eebd6f4e
Content-Type: application/json
Content-Length: 206

{
    "results": [
        {
            "code": 200,
            "package": "foo.bar",
            "stage": "3b388863-fda5-4e30-bd77-d2c7f976d2ad",
            "status": "Stage deleted."
        }
    ]
}
```

**Pull request merged**